### PR TITLE
Fix XML/YAML config loading

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/A
| License       | MIT
| Doc PR        | n/a

Fix an error when the `Resources/config` directory doesn't exist in the bundle (`[InvalidArgumentException] The "/api-platform/src/AppBundle/Resources/config/" directory does not exist.`).

Improve the code to load XML and YAML files.
                          
ping @soyuka

### CODE REVIEW

File: src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php [hunk 1]
```diff
@@ -149,23 +149,23 @@ private function registerAnnotationLoaders(ContainerBuilder $container)
...
-            }
-
-            $xmlResources = array_merge($xmlResources, glob($configDirectory.'api_resources.xml'));
-
-            foreach (Finder::create()->files()->in($configDirectory)->path('api_resources')->name('*.xml') as $file) {
-                $xmlResources[] = $file->getRealPath();
+            $configDirectory = dirname($reflectionClass->getFileName()).'/Resources/config/';
+
+            try {
+                foreach (Finder::create()->files()->in($configDirectory)->path('api_resources')->name('*.{yml,yaml,xml}') as $file) {
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php]: `api_resources.{yml,yaml,xml}` doesn't seem to be loaded anymore?

**Comment by soyuka** [comment]
see #576 too, take the want you prefer but anyway both implementations are corerct ;)

File: src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php (hunk 1 above)
Comment (soyuka) [inline src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php]: Behat is requiring both to be loaded to pass if I remember correctly. If it's not the case, there's a missing test case.

**Comment by dunglas** [comment]
I prefer this implementation because it doesn't rely on `glob` + the Symfony Finder but only on the finder.

**Comment by soyuka** [comment]
Can you explain how it takes `api_resources.xml` and `api_resources/*.xml`? Is `Finder::path` matching directories with files?

**Comment by dunglas** [comment]
It doesn't work. A test should be added for that.

**Comment by dunglas** [comment]
My bad, the test (and my code) work fine.

**Comment by soyuka** [comment]
https://github.com/api-platform/core/blob/master/features/configurable.feature 

Both are there I checked so this can be merged ;). 

Fun that `->path` takes both into consideration, good to know.

**Comment by dunglas** [comment]
Thank for your help @soyuka

**Comment by teohhanhui** [comment]
@soyuka:

> Fun that ->path takes both into consideration, good to know. Hope it's a feature and not a bug.

Perhaps we should open an issue on symfony/symfony. It's not clear from the documentation... http://symfony.com/doc/current/components/finder.html#path

**Pull request merged**